packages/dds/tree/src/simple-tree/schemaFactoryRecursive.ts

```diff
@@ -151,3 +152,34 @@ export type ValidateRecursiveSchema<
 		}[T["kind"]]
 	>,
 > = true;
+
+/**
+ * Workaround for fixing errors an issue with recursive ArrayNode schema exports.
+ * @remarks
+ * Importing a recursive ArrayNode schema via a d.ts file can produce an error like
+ * `error TS2310: Type 'RecursiveArray' recursively references itself as a base type.`
+ * if using a tsconfig with `"skipLibCheck": false`.
+ *
+ * This error occurs due to the TypeScript compiler splitting the class definition into two separate declarations in the d.ts file (one for the base, and one for the actual class).
+ * For unknown reasons, splitting the class declaration in this way breaks the recursive type handling, leading to the mentioned error.
+ *
+ * Since recursive type handling in TypeScript is order dependent, putting just the right kind of usages of the type before the declarations can cause it to not hit this error.
+ * For the case of ArrayNodes, this can be done via usage that looks like this:
+ *
+ * @example
+ * This example should use a doc comment to ensure the workaround comment shows up in the intellisense for the dummy export,
+ * however doing so is impossible due to how this example is included in a doc comment.
+ * ```typescript
+ *  // Workaround to avoid
+ *  // `error TS2310: Type 'RecursiveArray' recursively references itself as a base type.` in the d.ts file.
+ * export declare const _RecursiveArrayWorkaround: FixRecursiveArraySchema<typeof RecursiveArray>;
+ * export class RecursiveArray extends schema.arrayRecursive("RA", [() => RecursiveArray]) {}
+ * {
+ * 	type _check = ValidateRecursiveSchema<typeof RecursiveArray>;
+ * }
+ * ```
+ *
+ * This type always evaluates to undefined to ensure the dummy export (which doesn't exist at runtime) is typed correctly.
```

Comment:
Presumably this note wants to go in the `@remarks`, rather than the `@example` block.

Also nit:
```suggestion
 * This type always evaluates to `undefined` to ensure the dummy export (which doesn't exist at runtime) is typed correctly.
```